builder/frameworks/arduino.py

```
@@ -28,6 +28,10 @@
 import os
 import sys
 import shutil
+import hashlib
+from functools import lru_cache
+from pathlib import Path
+from typing import Dict, List, Tuple, Optional
```

Ruff (0.11.9)

`typing.Optional` imported but unused

Remove unused import: `typing.Optional`

(F401)